### ISSUE TITLE
Support pivoting on OS platform and Architecture for image selection

### ISSUE THREAD

**Issue by baronfel** [dotnet/sdk-container-builds#91]
Right now we explicitly support Linux/x64 container images - we should instead take into account the OS and Architecture the user provides and pick the correct 'base image' from the manifest list (if there is one on the base image).

We'll need to investigate what level of impact this has on picking the correct base image, but this will also have an impact on
* WorkingDirectory
* EntryPoint

Overlaps with https://github.com/dotnet/sdk-container-builds/issues/117

**Comment by bgiromini** [comment]
I assume this is still not supported with the release of dotnet 7 today?

**Comment by rainersigwald** [comment]
@bgiromini correct--the 0.2 version of the package is still `linux-x64` only. But this is definitely still on the to-do list!

**Comment by MrRickwi** [comment]
I tried to use debian.11 as OS - which is still Linux - but didn't work. Would be nice to see at least support for all Linux-like OSs in very near future - because we have packages in our project which have nothing in runtimes/linux-x64, but e.g. runtimes/debian.11-x64

**Comment by baronfel** [comment]
Thanks for the feedback @MrRickwi - I'm working on this in #247 and one of the open questions I was thinking about was 'how seriously do we need to handle RID support/compatibility'? Your response makes me think we need to fully support it.

**Comment by Zigby** [comment]
I was hoping it would work to publish arm images like this 
```
dotnet publish --os linux --arch arm64 

```
Is it correct that this issue will address publishing linux-arm64?

**Comment by Zigby** [comment]
If I understood correctly I should be able to create arm64 images by explicitly specifying the ContainerBaseImage. I cannot get it to work so here is what I am doing ...
I am omitting the architecture from the command line
```dotnet publish --os linux -p:PublishProfile=DefaultContainer```
and specify the container image in the csproj file (just picked what I thought would be appropriate)
```
 <ContainerBaseImage>mcr.microsoft.com/dotnet/aspnet:7.0-bullseye-slim-arm64v8</ContainerBaseImage>
```
I get the following error message
``` The Microsoft.NET.Build.Containers package does not (yet) support publishing for linux-arm64 ```

If I specify "--arch x64" it produces an image but it does not properly run on my arm computer.

Do I need to wait for a fix, or am I doing it wrong?

**Comment by baronfel** [comment]
@Zigby you're exactly right - I forgot we explicitly locked that down for the initial release [here](https://github.com/dotnet/sdk-container-builds/blob/main/packaging/build/Microsoft.NET.Build.Containers.targets#L26-L31). My apologies! So yes, you're blocked until we release #247.

**Comment by Zigby** [comment]
Thanks for the clarification and prompt response. I will stick to docker files for a little longer...

**Comment by baronfel** [comment]
We're currently planning to [freeze the RID graph](https://github.com/dotnet/designs/pull/260) and make RIDs no longer distro-specific. As a result, we're not going to go to full detail in this project to support e.g. `ubuntu22.04-x64` for example. Instead, for publishing containers you'd specify `linux-x64` (or `linux-arm64` or another supported RID for the container manifest list).  If a RID is specified that there is not match for on the discovered manifest list, an error will be raised specifying the valid RIDs for the manifest list.

**Comment by baronfel** [comment]
Fixed in #247, out now in version 0.3.2 of the package!
